### PR TITLE
README fixes/enhancements

### PR DESCRIPTION
**Pull request by chipzoller**
This PR:

* Adds yaml fencing to code blocks.
* Fixes some misspelling and Markdown linting issues.
* Adds a note on MPS experimental status.
* Adds a paragraph on sharing scope.
* Converts some lines to [GitHub Alerts Markdown syntax](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts).
* Makes a correction regarding how GFD comes packaged (i.e., no longer as a subchart as of v0.15.0).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -892,7 +918,7 @@ the `gpu-feature-discovery` component in standalone mode.
 
 The most basic installation command without any options is then:
 
-```
+```sh
```
Comment (elezar) [inline README.md:921]: we use `shell` elsewhere in this document, could we use that here instead?

File: README.md [hunk 2]
```diff
@@ -664,7 +688,8 @@ $ helm upgrade -i nvdp nvdp/nvidia-device-plugin \
 For multiple config files, the procedure is similar.
 
 Create a second `config` file with the following contents:
-```
+
+```yaml
```
Comment (elezar) [inline README.md:692]: Only part of this block is yaml. Strictly speaking this is `shell`.

File: README.md (hunk 1 above)
Comment (chipzoller) [reply README.md:921]: Sure

File: README.md (hunk 2 above)
Comment (chipzoller) [reply README.md:692]: Yes, but the majority of the contents are YAML and that syntax highlights the contents the best, but happy to change if desired.

Comment (ArangoGutierrez) [reply README.md:692]: Hey @chipzoller , thanks for the PR,  Please change it to yaml

Comment (chipzoller) [reply README.md:692]: It is currently set to yaml. Did you mean shell?

Comment (ArangoGutierrez) [reply README.md:692]: Ups yeah shell

Comment (chipzoller) [reply README.md]: Done

**Review by ArangoGutierrez — COMMENTED** [review]
lgtm

**Comment by chipzoller** [comment]
Thanks for the review. This look good to go?

**Comment by chipzoller** [comment]
Since this is still open, I added a short paragraph about the scope of sharing.

File: README.md [hunk 3]
```diff
@@ -337,10 +352,15 @@ In contrast to time-slicing, MPS does space partitioning and allows memory and
 compute resources to be explicitly partitioned and enforces these limits per
 workload.
 
+With either time-slicing or MPS, the same sharing method is applied to all GPUs on
```
Comment (elezar) [inline README.md:355]: Nit:
```suggestion
With both time-slicing and MPS the same sharing method is applied to all GPUs on
```

File: README.md [hunk 4]
```diff
@@ -747,7 +776,8 @@ direct values to set the configuration options of the plugin without using a
 `ConfigMap`), or used to override these options as desired.
 
 These values are as follows:
-```
+
+```yaml
```
Comment (elezar) [inline README.md:780]: This isn't strictly yaml. Could you provide an example of how this renders?

File: README.md [hunk 5]
```diff
@@ -4,32 +4,45 @@
...
-      - [Multiple Config File Example](#multiple-config-file-example)
-      - [Updating Per-Node Configuration With a Node Label](#updating-per-node-configuration-with-a-node-label)
-    + [Setting other helm chart values](#setting-other-helm-chart-values)
-    + [Deploying with gpu-feature-discovery for automatic node labels](#deploying-with-gpu-feature-discovery-for-automatic-node-labels)
-    + [Deploying gpu-feature-discovery in standalone mode](#deploying-gpu-feature-discovery-in-standalone-mode)
-  * [Deploying via `helm install` with a direct URL to the `helm` package](#deploying-via-helm-install-with-a-direct-url-to-the-helm-package)
-- [Building and Running Locally](#building-and-running-locally)
-- [Changelog](#changelog)
-- [Issues and Contributing](#issues-and-contributing)
+- [NVIDIA device plugin for Kubernetes](#nvidia-device-plugin-for-kubernetes)
```
Comment (elezar) [inline README.md:7]: Question: What is the intent of this top-level element? It doesn't really seem to add value from a text / navigation perspective.

File: README.md [hunk 6]
```diff
@@ -614,7 +641,8 @@ that have not been customized via a node label (more on this later).
 
 #####  Single Config File Example
 As an example, create a valid config file on your local filesystem, such as the following:
-```
+
+```yaml
```
Comment (elezar) [inline README.md:645]: why is this not `shell` when the example on line 696 is?

**Review by elezar — COMMENTED** [review]
Thanks @chipzoller.

I think these change look good from a content perspective. Definitely some good additions and clarifications.

I would like for the types on the code blocks to be used consistently though and would like to better understand the TOC changes.

File: README.md (hunk 5 above)
Comment (chipzoller) [reply README.md:7]: It was some auto-format work that snuck in inadvertently. Removed.

File: README.md (hunk 4 above)
Comment (chipzoller) [reply README.md:778]: See example [here](https://github.com/chipzoller/nvidia-k8s-device-plugin/tree/cz-readme-fixes?tab=readme-ov-file#setting-other-helm-chart-values).

File: README.md (hunk 6 above)
Comment (chipzoller) [reply README.md:645]: Standardized.

**Comment by chipzoller** [comment]
@elezar anything else here or can we merge?

**Comment by chipzoller** [comment]
Bump

**Comment by chipzoller** [comment]
Folks, this simple readme PR has been outstanding for a month. What more can I do to help update your documentation here?

**Comment by mikemckiernan** [comment]
> Folks, this simple readme PR has been outstanding for a month. What more can I do to help update your documentation here?

Thanks for all your time and help.  If you don't object to my suggs and requests, but don't have time, PLMK if it's OK to push a commit.

**Comment by chipzoller** [comment]
Sure, and I'm glad to implement any suggestions as well.

**Review by mikemckiernan — CHANGES_REQUESTED** [review]
Thanks again for your willingness to pitch in.

**Comment by chipzoller** [comment]
Done

**Review by mikemckiernan — APPROVED** [review]
Thanks again for the assist.

**Comment by mikemckiernan** [comment]
We'll look at this after the lint CI job is revised.

**Comment by chipzoller** [comment]
Merged main so CI can probably be kicked off again assuming it pulled in those fixes.

**Pull request merged**